endpointmanager/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -148,6 +173,7 @@ func saveMsgInDB(message []byte, args *map[string]interface{}) error {
 			existingEndpt.HTTPResponse = fhirEndpoint.HTTPResponse
 		}
 		existingEndpt.Errors = fhirEndpoint.Errors
+		existingEndpt.Validation = fhirEndpoint.Validation
```

Comment:
For the code above - 
let's always replace these values rather than check if they are valid. that will make sure that the existing snapshot is accurate for the current state - if the capability statement didn't get returned, it should be removed.

long term, we'll have a way of maintaining history. For now, this will help the db remain consistent.